internal/integration/unified/crud_helpers.go

```diff
@@ -20,65 +22,83 @@ func newMissingArgumentError(arg string) error {
 	return fmt.Errorf("operation arguments document is missing required field %q", arg)
 }
 
-type updateArguments struct {
+type updateArguments[Options options.UpdateManyOptions | options.UpdateOneOptions] struct {
 	filter bson.Raw
 	update interface{}
-	opts   *options.UpdateOptionsBuilder
+	opts   options.Lister[Options]
 }
 
-func createUpdateArguments(args bson.Raw) (*updateArguments, error) {
-	ua := &updateArguments{
-		opts: options.Update(),
+func createUpdateArguments[Options options.UpdateManyOptions | options.UpdateOneOptions](
```

Comment:
We should not try to generalize this function. Instead, we should maintain `createUpdateOneArguments` and `createUpdateManyArguments`. Using the combination of reflection and generics here is confusing and difficult to read.